tests/unit/htdocs/class/logger/XoopsFileLoggerTest.php

```
@@ -0,0 +1,456 @@
+            foreach (glob($live . '.*') ?: [] as $candidate) {
+                if (is_file($candidate) && file_get_contents($candidate) === 'ROTATION-2') {
+                    $survived = true;
+                }
+            }
+            self::assertTrue($survived, 'the rotation that could not move must not be overwritten');
+            self::assertTrue($this->property($logger, 'writeFailed'), 'a stalled rotation disables the logger');
+        } finally {
+            @unlink($blocked . '/blocker');
+            @rmdir($blocked);
+        }
+    }
```

PHPMD (2.15.0)

[error] Remove error control operator '@' on line 339. (undefined)

(ErrorControlOperator)